aws/resource_aws_ecr_repository_test.go

```diff
@@ -2,15 +2,64 @@ package aws
 
 import (
 	"fmt"
+	"log"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ecr"
+	multierror "github.com/hashicorp/go-multierror"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_ecr_repository", &resource.Sweeper{
+		Name: "aws_ecr_repository",
+		F:    testSweepEcrRepositories,
+	})
+}
+
+func testSweepEcrRepositories(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).ecrconn
+
+	var errors error
+	err = conn.DescribeRepositoriesPages(&ecr.DescribeRepositoriesInput{}, func(page *ecr.DescribeRepositoriesOutput, isLast bool) bool {
+		if page == nil {
+			return !isLast
+		}
+
+		for _, repository := range page.Repositories {
+			shouldForce := true
+			_, err = conn.DeleteRepository(&ecr.DeleteRepositoryInput{
+				// We should probably sweep repositories even if there are images.
+				Force:          &shouldForce,
+				RegistryId:     repository.RegistryId,
+				RepositoryName: repository.RepositoryName,
+			})
+			if err != nil {
```

Comment:
Please add

```go
if isAWSErr(err, ecr.ErrCodeRepositoryNotFoundException, "") {
    continue
}
```

just in case someone else has deleted the repository in the meantime.